### PR TITLE
cmd/operator-sdk: scaffold a boilerplate file for new and migrated projects

### LINKED ISSUE #1270: add option to generate the header files to all files and not only for api
## Feature Request

today looks like you can add a custom header file only for the api, will be good to have an option to add to all files in the project

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:** move `--header-file` from `generate {k8s,openapi}` to `new` and `migrate` commands. The header file data is copied to `hack/boilerplate.go.txt` and used in all future Go code scaffolding/codegen events.

**Motivation for the change:** consolidates functionality. See #1270 

Closes #1270

### CODE REVIEW

**Comment by estroz** [comment]
@LiliC fixed. Forgot to check the `CustomRenderer` interface spec :sweat_smile:.

**Comment by LiliC** [comment]
It works now! 👍  

For example, if the text is not commented out `//` in the boilerplate file the project initialization fails as `dep ensure` cannot run. Not sure if we should be doing any validation against the passed in boilerplate file, I guess it's users fault if this happens. WDYT?

**Comment by estroz** [comment]
@LiliC good idea, I'll add validation for comments.

File: internal/pkg/scaffold/scaffold.go [hunk 1]
```diff
@@ -73,6 +79,46 @@ func (s *Scaffold) configure(cfg *input.Config) {
...
+		return nil, err
+	}
+	if len(bytes.TrimSpace(b)) == 0 {
+		return nil, fmt.Errorf(`boilerplate file "%s" is empty`, bp)
+	}
+	scanner := bufio.NewScanner(bytes.NewBuffer(b))
+	for scanner.Scan() {
+		line := scanner.Text()
+		if !strings.HasPrefix(strings.TrimSpace(line), commentPrefix) {
+			return nil, fmt.Errorf(`this boilerplate line must be a comment: "%s"`, line)
```
Comment (LiliC) [inline internal/pkg/scaffold/scaffold.go]: The comment can also be a `/* */` block comment.

File: internal/pkg/scaffold/scaffold.go [hunk 2]
```diff
@@ -73,6 +80,60 @@ func (s *Scaffold) configure(cfg *input.Config) {
...
+			if _, err := os.Stat(i.Path); err == nil {
+				bp = i.Path
+			}
+		}
+		if bp != "" {
+			s.BoilerplateBytes, err = getAndValidateBoilerplateBytes(bp)
+			return err
+		}
+	}
+	return nil
```
[the remark this answers is not in the document]
Comment (estroz) [reply internal/pkg/scaffold/scaffold.go]: The easiest solution is to unexport. Pushing that change now.

**Pull request merged**